sphinx/builders/linkcheck.py

```diff
@@ -40,10 +40,11 @@
 
 uri_re = re.compile('([a-z]+:)?//')  # matches to foo:// and // (a protocol relative URL)
 
-Hyperlink = NamedTuple('Hyperlink', (('next_check', float),
-                                     ('uri', Optional[str]),
+Hyperlink = NamedTuple('Hyperlink', (('uri', Optional[str]),
                                      ('docname', Optional[str]),
```

Comment:
With the entire Hyperlink being optional, the threads can be shut down without passing `(None, None, None)` for `uri`, `docname` and `lineno`. That means `uri` and `docname` are no longer optional, right?

`HyperlinkAvailabilityCheckWorker.check()` body first checks `len(uri)`, so a uri of `None` would have broken linkcheck. Similarly, `is_ignore_uri` checks for a pattern in `uri` and would break if `uri` is None.

Reply:
Absolutely!